src/SFA.DAS.EmployerIncentives.Data/ApprenticeshipIncentives/ApprenticeshipIncentiveDataRepository.cs

```diff
@@ -151,6 +129,34 @@ public async Task Delete(ApprenticeshipIncentiveModel apprenticeshipIncentive)
             }
         }
 
+        private async Task<ApprenticeshipIncentiveModel> Get(Expression<Func<ApprenticeshipIncentive, bool>> predicate)
+        {
+            var apprenticeshipIncentive = await _dbContext.ApprenticeshipIncentives.FirstOrDefaultAsync(predicate);
```

Comment:
You could pull back the PendingPayments and ValidationResults at the same time as the ApprenticeshipIncentives here.    the query should still be efficient since its parent-table->child-table->child-table .  Is the subsequent includes that cause the problems.

Reply:
Updated